middle_end/flambda2/to_cmm/to_cmm_shared.ml

```diff
@@ -213,10 +213,9 @@ let const_static cst =
            (tag_targetint (Targetint_31_63.to_targetint i))) ]
   | Naked_float f -> [cfloat (Numeric_types.Float_by_bit_pattern.to_float f)]
   | Naked_float32 f ->
-    (* Here we are relying on the data section being zero initialized to
-       maintain the invariant that statically-allocated float32 values are zero
-       padded. *)
-    [cfloat32 (Numeric_types.Float32_by_bit_pattern.to_float f)]
+    (* Statically-allocated float32 values are zero padded. We must explicitly
+       add the padding otherwise subsequent values will be misaligned. *)
+    [cfloat32 (Numeric_types.Float32_by_bit_pattern.to_float f); cint32 0l]
   | Naked_int32 i -> [cint (Nativeint.of_int32 i)]
```

Comment:
I wouldn't mind changing the `Naked_int32` case to use two `cint32` instead of one `cint`. It is more obviously correct and does not depend on endianness.

Reply:
Agreed, but let's do this in a separate PR